### PR TITLE
Flag images without alt text

### LINKED ISSUE #23: Flag images with no alt text at all
I really like the idea behind this action! However, the first few images I tried to test it with didn’t get flagged like I expected.

In the first instance, I happened to drag an image into the text area that had a filename that wasn’t “Screenshot <date>” (it was “edit-link.png”). I’m not sure if there’s a good way to identify stuff like this (maybe check alt-text with no spaces against a dictionary? tough to handle lots of languages, though…).

In the second instance, I tried just deleting the alt text entirely (all the variations of ``, `` (no alt attribute at all), and ``). These seem more obviously straightforward to identify, and it would be nice if they were caught by this action.

---

As a side note, I thought I’d configured something wrong because of the above. The first thing I did was look at the action’s logging output, and it would have been a little bit helpful to have the two outputs (whether an image was flagged and the type of thing being checked) labeled so I knew what they were: https://github.com/github/accessibility-alt-text-bot/blob/b8484e124855b2e20bf7d01dd7fcdc2d536ab906/action.yml#L62-L63

(e.g. `echo "Detected bad alt text: ${flag}"` would have been helpful.)

### PR DESCRIPTION
**Pull request by kendallgassner**
resolves #23 

### What

I am adding a couple new flags:
- Flags when an image tag contians ``

Our CI was not actually failing when a test failed. I fixed this by adding `exit 1` when a unit test failed.

### CODE REVIEW

File: flag-alt-text.sh [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
-    echo false
-  fi
+    markdownMacOsScreenshotRegex="^.*!\[(Clean|Screen) ?[S|s]hot [0-9][0-9][0-9][0-9]-[0-1][0-9]-[0-9][0-9].*\].*$"
+    semanticMacOsScreenshotRegex="^.*<img.*(Clean|Screen) ?[S|s]hot [0-9][0-9][0-9][0-9]-[0-1][0-9]-[0-9][0-9].*$"
+    markdownImageRegex="^.*!\[(i|I)mage\].*$"
+    semanticImageRegex="^.*<img.*alt=\"(i|I)mage\".*$"
+    emptySemanticRegex="^.*<img.*alt=(\"|')(\"|').*$"
+    emptyMarkdownRegex="^.*!\[\].*$"
+
+    if [[ $1 =~ $semanticMacOsScreenshotRegex ]] || [[ $1 =~ $markdownMacOsScreenshotRegex ]] || [[ $1 =~ $semanticImageRegex ]] || [[ $1 =~ $markdownImageRegex ]] || [[ $1 =~ $emptySemanticRegex ]] || [[ $1 =~ $emptyMarkdownRegex ]]; then
```
Comment (khiga8) [inline flag-alt-text.sh:11]: Could we possibly check if something IS an HTML img, but doesn't have `alt=` using the NOT operator?

Syntax might be off, but something like:
```
    semanticImageRegex="^.*<img.*$"
    semanticImageWithAltRegex="^.*<img.*alt=.*$"
```
then in the if conditional we have:

```
|| [[ ($1 =~ $semanticImageRegex) && !($1 =~ $semanticImageWithAltRegex) ]]
```

File: action.yml [hunk 2]
```diff
@@ -59,8 +59,8 @@ runs:
           Learn more about alt text at [Basic writing and formatting syntax: images on GitHub Docs](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#images)."
           flag="$(flagAltText "$content")"
           
-          echo $flag
-          echo $type
+          echo "Detected bad alt text: ${flag}"
+          echo "Event type: $type"
```
Comment (khiga8) [inline action.yml:63]: Nice and clear!

File: flag-alt-text.sh (hunk 1 above)
Comment (kendallgassner) [reply flag-alt-text.sh:11]: Love this idea let me throw it in

**Review by khiga8 — APPROVED** [review]
Nice work 🎉

**Pull request merged**